### PR TITLE
Add rabbitmq support

### PR DESCRIPTION
**Pull request by victorcastanheiro-luizalabs**
This PR adds support to RabbitMQ on Mitose.

It is basically a copy-paste from the SQS backend using the number of messages provided by RabbitMQ api.

Checklist
---
- [x] Create rabbitmq backend
- [x] Create rabbitmq controller
- [x] Update readme
- [ ] Tests
- [ ] Changelog

### CODE REVIEW

File: rabbitmq/rabbitmq.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Basic %s", credentials))
+
+	res, err := client.Do(req)
+	if err != nil {
+		return -1, err
+	}
+	defer res.Body.Close()
+
+	var result map[string]interface{}
```
Comment (drgarcia1986) [inline rabbitmq/rabbitmq.go:27]: What about a struct with the desired fields? f.ex:
```go
type RQResponse struct {
    Messages float64 `json: messages`
}
```

**Review by drgarcia1986 — APPROVED** [review]
LGTM 👍

File: rabbitmq/rabbitmq.go (hunk 1 above)
Comment (victorcastanheiro-luizalabs) [reply rabbitmq/rabbitmq.go:27]: Make sense!

Doing 👀

Comment (victorcastanheiro-luizalabs) [reply rabbitmq/rabbitmq.go]: Done!

@drgarcia1986 can you check again pls? 🙏

**Review by viniciusfca — APPROVED** [review]
:+1:

File: rabbitmq/rabbitmq.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	res, err := client.Do(req)
+	if err != nil {
+		return -1, err
+	}
+	defer res.Body.Close()
+
+	rabbitMQResponse := new(RabbitMQResponse)
+
+	err = json.NewDecoder(res.Body).Decode(&rabbitMQResponse)
```
Comment (drgarcia1986) [inline rabbitmq/rabbitmq.go:33]: the `rabbitMQResponse` variable is already a pointer, do you need to use `&` here?

**Comment by drgarcia1986** [comment]
Hey @vinicyusmacedo could you review?

File: rabbitmq/rabbitmq.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+type RabbitMQClient struct {
+}
+
+type RabbitMQResponse struct {
+	Messages float64 `json:"messages"`
+}
+
+func (r *RabbitMQClient) GetNumOfMessages(url, credentials string) (int, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, nil)
```
Comment (vinicyusmacedo) [inline rabbitmq/rabbitmq.go:18]: Is the URL the same as the amqp endpoint? (the one that connects to port 5672)

Comment (victorcastanheiro-luizalabs) [reply rabbitmq/rabbitmq.go:18]: No is not the same, this is the URL of the RabbitMQ console endpoint (port 15672)

Comment (vinicyusmacedo) [reply rabbitmq/rabbitmq.go:18]: Oh, cool. Do you know which permissions are needed for that?

**Review by retpolanne — APPROVED** [review]

File: rabbitmq/rabbitmq.go (hunk 3 above)
Comment (victorcastanheiro-luizalabs) [reply rabbitmq/rabbitmq.go:18]: A basic user with read permission on queue can obtain the information provided by api.

**Pull request merged**